client.go

```diff
@@ -223,14 +228,32 @@ func (c *Client) recvVersion() error {
 		return &unexpectedPacketErr{sshFxpVersion, typ}
 	}
 
-	version, _ := unmarshalUint32(data)
+	version, data := unmarshalUint32(data)
 	if version != sftpProtocolVersion {
 		return &unexpectedVersionErr{sftpProtocolVersion, version}
 	}
 
+	for len(data) > 0 {
+		var ext extensionPair
+		ext, data, err = unmarshalExtensionPair(data)
+		if err != nil {
+			return err
+		}
+		c.ext[ext.Name] = ext.Data
+	}
+
 	return nil
 }
 
+// HasExtension checks whether the server supports a named extension.
+//
+// The second return value is the extension data reported by the server
+// (typically a version number).
+func (c *Client) HasExtension(name string) (ok bool, data string) {
+	data, ok = c.ext[name]
```

Comment:
Reversing the arguments returned from the map assignment probably violates the Principle of Least Astonishment.

Also, the naming of the arguments is probably unimportant. We should be able to know what the two fields are, especially when they are `string, bool`. The “comma ok” idiom is strong in Go.

Reply:
Reversed and removed the return value names.